File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(string s) {
    string answer = "";
    
    bool is_first = true;
    
    for(int i=0; i<s.length();i++){
        if(is_first){
            // 첫번째 문자면
            s[i] = toupper(s[i]);
            answer += s[i];
        } else {
            s[i] = tolower(s[i]);
            answer += s[i];
        }
        
        if(s[i] == ' '){
            is_first = true;
        } else {
            is_first = false;
        }
    }
    
    
    return answer;
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
import java.util.*;

class Solution {
    public String solution(String s) {
        String answer = "";
        
        String[] words = s.toLowerCase().split("");
        
        boolean isFirst = true;
        
        for(int i=0; i<words.length;i++){
            if(isFirst){
                answer += words[i].toUpperCase();
            } else {
                answer += words[i];
            }
            
            if(words[i].equals(" ")){
                isFirst = true;
            } else {
                isFirst = false;
            }
        }
        
        
        
        
        return answer;
    }
}